src/org/thoughtcrime/securesms/ApplicationContext.java

```diff
@@ -112,6 +112,17 @@ public void onCreate() {
       }
       Log.i("DeltaChat", "shutting down event handler");
     }, "eventThread").start();
+
+    // set translations before starting I/O to avoid sending untranslated MDNs (issue #2288)
+    DcHelper.setStockTranslations(this);
+    IntentFilter filter = new IntentFilter(Intent.ACTION_LOCALE_CHANGED);
```

Comment:
we could still leave `registerReceiver` at the old place and move only the initial call to `setStockTranslations()` up - that would result in I/O starting sooner and running in parallel with registering receiver with a filter (not sure, however, how expensive that really is, but there may be slow systems and we were surprised by how slow android calls are here and there :)

also, leaving `registerReceiver` would make the change smaller, on startup things are sometimes critical wrt ordering (things may not be initialized etc.) however, i do not see side effects here (which does not mean that they cannot be there :)

therefore, _i_ would probably have changed only the single line - but i also see that it is nice to keep things together.

Reply:
makes sense, done